src/bootloaders/shim-systemd.c

```diff
@@ -110,7 +111,7 @@ static char *shim_systemd_get_kernel_dst(const BootManager *manager) {
 
 static bool shim_systemd_install_kernel(const BootManager *manager, const Kernel *kernel) {
         fprintf(stderr, "Call %s\n", __func__);
-        return true;
+        return sd_class_install_kernel_impl(manager, kernel, shim_systemd_get_kernel_dst, NULL);
```

Comment:
I do not have a strong opinion, but I do have an opinion.

1. I've defined layout in such a way that everything that we touch is under root `/org.clearlinux`.  We don't go any other place and there's no confusion as to where are the files that we manage.  systemd's hard-coded `/loader` is unfortunate exception, but it's life (and systemd, too).

2. Conceptually, each bootloader's implementation in `src/bootloader` is the owner of its own layout. It's not only tied to kernels, but also bootloaders themselves, their names and location. I do not think that a part of a booting scheme should be defined elsewhere. I've reused creation of conf entry for systemd-boot.

These were my reasons for doing it this way. Now, we can make the naming schemes coincide. However, I do not see how it is helpful.

Just to make sure we're on the same page: this is a replacement scheme for the current one. We will not be using them in parallel. Therefore, we will be no two location to manage. It's not that we're going to be maintaining two layouts depending on whether secure boot is enabled or not. But...

There's been a change in layout (in June, I guess) and kernels were moved. That change simply left stray kernels in their previous location. Which is not nice.

If your concern behind the location of the kernels is managing the transition, then we can make sure there's a clean up code in place that will ensure no artifacts of the previous layout are left on the ESP (i.e. kernels and bootloaders are removed).

So, I'm suggesting to let a bootloader (or bootscheme) auto-contain the definition of the layout. Let it all be in `/org.clearlinux`. Additional item as the result of this discussion would be adding code that cleans up the ESP. Would that be OK by you?